### PR TITLE
Refactor handler argument calling

### PR DESCRIPTION
**Pull request by MatthewScholefield**
## Description
This simplifies the logic in `add_event`, specifically for calling the handler. It also removes the `has_self` parameter of `add_event` which could break any skills that use it with that parameter.

## How to test
Test if skills work with this change

### CODE REVIEW

File: mycroft/skills/core.py [hunk 1]
```diff
@@ -591,45 +590,21 @@ def wrapper(message):
...
                 # Convert "MyFancySkill" to "My Fancy Skill" for speaking
-                handler_name = re.sub("([a-z])([A-Z])", "\g<1> \g<2>",
-                                      self.name)
-                # TODO: Localize
-                self.speak("An error occurred while processing a request in " +
-                           handler_name)
-                LOG.error("An error occurred while processing a request in " +
-                          self.name, exc_info=True)
+                handler_name = re.sub(r"([a-z])([A-Z])", r"\1 \2", self.name)
+                data = {'skill': handler_name}
```
Comment (forslund) [inline mycroft/skills/core.py]: This overwrites the data prepared previously for the skill.handler.started / .stopped messages (the `name` key will be slightly different)

File: mycroft/skills/core.py [hunk 2]
```diff
@@ -591,45 +590,21 @@ def wrapper(message):
...
-                            try:
-                                handler(self, unmunge_message(message,
-                                                              self.skill_id))
-                            except TypeError:
-                                handler(self)
-                        else:
-                            LOG.error("Unexpected argument count:" +
-                                      str(len(getargspec(handler).args)))
-                            raise TypeError
+                    is_bound = bool(getattr(handler, 'im_self', None))
```
Comment (forslund) [inline mycroft/skills/core.py]: May possibly be simplified slightly by using 
`is_bound = hasattr('im_self')`

Comment (MatthewScholefield) [reply mycroft/skills/core.py]: From what I understand, if a class method is given accessed statically via `ClassName.function`, `im_self` will exists but be `None`.

Comment (forslund) [reply mycroft/skills/core.py]: Ah, right you are. Would fail for classmethods

File: mycroft/skills/core.py (hunk 1 above)
Comment (MatthewScholefield) [reply mycroft/skills/core.py]: Nice catch! Fixed.

**Comment by forslund** [comment]
Excellent! Merging!

**Pull request merged**